### PR TITLE
sync with AnacondaRecipes

### PR DESCRIPTION
**Pull request by bgruening**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy`
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by bgruening** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,58 +1,39 @@
...
-    - glpk
-    - suitesparse 4.5.6|4.5.6.*   # [not win]
-    - suitesparse 5.1.*   # [win]
 
 test:
   imports:
     - cvxopt
   source_files:
-    - tests/*
+    - tests/*.py
```
Comment (martinandersen) [inline recipe/meta.yaml]: At least one test depends on a data file (`boeing2.mps`) in the `tests` directory.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,58 +1,39 @@
...
     - examples/*
-  requires:
-    - pytest
```
Comment (martinandersen) [inline recipe/meta.yaml]: `pytest` is used in `run_test.py`

File: recipe/meta.yaml (hunk 1 above)
Comment (bgruening) [reply recipe/meta.yaml]: I don't get how the AnacondaRecipes passes the tests :(

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,49 +1,32 @@
...
-
+    - dsdp
+    - suitesparse    # [unix]
   run:
     - python
-    - blas 1.1 {{ variant }}
-    - openblas
-    - gsl
-    - fftw
     - dsdp
```
Comment (martinandersen) [inline recipe/meta.yaml]: Is this implicit when included under host?

File: recipe/bld.bat [hunk 4]
```diff
@@ -1,29 +1,35 @@
...
+   set CC=c99wrap
+   )
 
-set "CVXOPT_SUITESPARSE_LIB_DIR=%LIBRARY_PREFIX%\lib"
-set "CVXOPT_SUITESPARSE_INC_DIR=%LIBRARY_PREFIX%\include\suitesparse"
+set "CVXOPT_MSVC=1"
 
 %PYTHON% setup.py install --single-version-externally-managed --record=record.txt
-
-copy src\C\cvxopt.h %LIBRARY_PREFIX%\include
```
[the remark this answers is not in the document]
Comment (martinandersen) [reply recipe/bld.bat]: Dropped it or never added it? :wink: We added it a few months ago.

The AnacondaRecipes version does not appear to include windows either. You'll need to include the patch for the windows build to work.

File: recipe/meta.yaml [hunk 5]
```diff
@@ -1,49 +1,35 @@
...
+    - openblas-devel {{ openblas }}  # [blas_impl == "openblas"]
     - gsl
     - fftw
-    - dsdp
     - glpk
-    - suitesparse 4.5.6|4.5.6.*   # [not win]
-    - suitesparse 5.1.*   # [win]
-
+    - dsdp
+    - suitesparse    # [unix]
```
Comment (martinandersen) [inline recipe/meta.yaml]: This  shouldn't be unix only.

File: recipe/meta.yaml [hunk 6]
```diff
@@ -1,49 +1,35 @@
...
-
 package:
   name: cvxopt
   version: {{ version }}
 
 source:
-  fn: cvxopt-{{ version }}.tar.gz
   url: https://pypi.io/packages/source/c/cvxopt/cvxopt-{{ version }}.tar.gz
   sha256: 3296c9d49b7dcb894b20db5d7d1c1a443912b4d82358e03f836575e8398e0d60
-  patches: cvxopt_win_patch.diff   # [win]
```
Comment (martinandersen) [inline recipe/meta.yaml]: This patch is currently necessary on Windows to build CVXOPT with FFTW and DSDP.

File: recipe/bld.bat [hunk 7]
```diff
@@ -1,29 +1,35 @@
...
+set "CVXOPT_FFTW_LIB_DIR=%LIBRARY_LIB%"
+set "CVXOPT_FFTW_INC_DIR=%LIBRARY_INC%"
 
 set CVXOPT_BUILD_GLPK=1
-set "CVXOPT_GLPK_LIB_DIR=%LIBRARY_PREFIX%\lib"
-set "CVXOPT_GLPK_INC_DIR=%LIBRARY_PREFIX%\include"
+set "CVXOPT_GLPK_LIB_DIR=%LIBRARY_LIB%"
+set "CVXOPT_GLPK_INC_DIR=%LIBRARY_INC%"
 
-set CVXOPT_BUILD_DSDP=1
```
Comment (martinandersen) [inline recipe/bld.bat]: The DSDP extension will not be included without this part.

File: recipe/meta.yaml (hunk 5 above)
Comment (jakirkham) [reply recipe/meta.yaml]: Also the pinnings above were needed to deal with Windows and Unix separately as the same versions are not available for both.

File: recipe/meta.yaml [hunk 8]
```diff
@@ -1,49 +1,35 @@
...
     - python
-    - blas 1.1 {{ variant }}
-    - openblas
+    - dsdp
     - gsl
     - fftw
-    - dsdp
     - glpk
-    - suitesparse 4.5.6|4.5.6.*   # [not win]
-    - suitesparse 5.1.*   # [win]
```
Comment (jakirkham) [inline recipe/meta.yaml]: Still need these in `run` with version constraints.

File: recipe/meta.yaml [hunk 9]
```diff
@@ -1,49 +1,35 @@
...
-    - {{ compiler('c') }}
+    - {{ compiler("c") }}
+    - c99-to-c89          # [win and vc<14]
   host:
     - python
     - setuptools
-    - blas 1.1 {{ variant }}
-    - openblas
+    - mkl-devel  {{ mkl }}  # [blas_impl == "mkl"]
+    - openblas-devel {{ openblas }}  # [blas_impl == "openblas"]
```
Comment (jakirkham) [inline recipe/meta.yaml]: Think we need to figure out how we deal with `blas` globally in conda-forge first. Then we can revisit updating this recipe according to that global spec.

**Pull request closed without merging**